src/pages/RegisterPage.jsx

```diff
@@ -1,7 +1,130 @@
-import React from "react";
+import React from 'react';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import { Link } from 'react-router-dom';
+import { toast } from 'react-hot-toast';
 
 const RegisterPage = () => {
-  return <div>RegisterPage</div>;
+  const handleSubmit = (values, { resetForm }) => {
+    const existingUser = JSON.parse(localStorage.getItem('user'));
+
+    if (existingUser && existingUser.email === values.email) {
+      toast.error('User already exists with this email!');
+    } else {
+      localStorage.setItem('user', JSON.stringify(values));
+      toast.success('Registration successful!');
```

Comment:
just a note i make a function to handle all showing toast in the app  it is like that 
showToast("Order deleted Successfully!", "success");